### ISSUE TITLE
add flatten layer

### ISSUE THREAD

**Issue by icodingc** [ethereon/caffe-tensorflow#31]
when i use kaffe-tensorflow to convert (Network in Network ) to tensorflow,but when i load the converted net ,i get error,because avg_pool.shape (batch ,10,1,1) have not been fatten，and the next layer is softmax.(batch,10) so i add a flatten layer in network.py
```
@layer
    def flatten(self,input,name):
        dim = 1
        for d in input.get_shape()[1:].as_list():
                dim *= d
        return tf.reshape(input,[-1,dim])
```

**Comment by ethereon** [comment]
Thanks for reporting this. It appears that the NCHW order in Caffe made the singleton dimensions broadcast-able for the softmax, but not TensorFlow's NHWC. The softmax layer now squeezes out the spatial dimensions, so an explicit flatten layer should be unnecessary (tested on the updated network in network model).

**Comment by yuhsh24** [comment]
after I add @icodingc code in network.py, I met the error message "Output shape computation not implemented for type: Flatten". How can I fix these error and convert caffe model which contains Flatten layers to tensorflow model sucessfully?

**Comment by stoneofcrazy** [comment]
@yuhsh24 Have you solved the problem? I had the same problem.
